src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,30 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+	public String value;
+	
+	protected boolean isPrivate;
+	
+	public Contact(String value, boolean isPrivate) {
+		this.value = value;
+		this.isPrivate = isPrivate;
+	}
+	
+	public String toString(){
+		return value;
+	}
+	
+	public boolean equals(Object other){
+		return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
```

Comment:
can you see why this is a problem? you can now potentially compare `Address` with `Email`, why?